src/vm/functions/assets.rs

```diff
@@ -0,0 +1,211 @@
+use std::convert::TryFrom;
+
+use vm::functions::tuples;
+use vm::functions::tuples::TupleDefinitionType::{Implicit, Explicit};
+
+use vm::types::{Value, OptionalData, BuffData, PrincipalData, BlockInfoProperty, AtomTypeIdentifier};
+use vm::representations::{SymbolicExpression};
+use vm::errors::{Error, UncheckedError, InterpreterError, RuntimeErrorType, InterpreterResult as Result, check_argument_count};
+use vm::{eval, LocalContext, Environment};
+
+pub fn special_mint_token(args: &[SymbolicExpression],
+                          env: &mut Environment,
+                          context: &LocalContext) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let token_name = args[0].match_atom()
+        .ok_or(UncheckedError::InvalidArgumentExpectedName)?;
+
+    let amount = eval(&args[1], env, context)?;
+    let to =     eval(&args[2], env, context)?;
+
+    if let (Value::Int(amount),
+            Value::Principal(ref to_principal)) = (amount, to) {
+        if amount < 0 {
+            return Err(RuntimeErrorType::TransferNonPositiveAmount.into())
+        }
+
+        let to_bal = env.global_context.database.get_token_balance(&env.contract_context.name, token_name, to_principal)?;
+
+        let final_to_bal = to_bal.checked_add(amount)
+            .ok_or(RuntimeErrorType::ArithmeticOverflow)?;
+
+        env.global_context.database.set_token_balance(&env.contract_context.name, token_name, to_principal, final_to_bal)?;
+
+        Ok(Value::okay(Value::Bool(true)))
+    } else {
+        Err(UncheckedError::InvalidArguments("mint-token! expects an integer amount and a to principal".to_string()).into())
+    }
+}
+
+// Error Codes => 1: asset already exists
+pub fn special_mint_asset(args: &[SymbolicExpression],
+                          env: &mut Environment,
+                          context: &LocalContext) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let asset_name = args[0].match_atom()
+        .ok_or(UncheckedError::InvalidArgumentExpectedName)?;
+
+    let asset =  eval(&args[1], env, context)?;
+    let to    =  eval(&args[2], env, context)?;
+
+    let expected_asset_type = env.global_context.database.get_asset_key_type(&env.contract_context.name, asset_name)?;
+
+    if !expected_asset_type.admits(&asset) {
+        return Err(UncheckedError::TypeError(expected_asset_type.to_string(), asset).into())
+    }
+
+    if let Value::Principal(ref to_principal) = to {
+        match env.global_context.database.get_asset_owner(&env.contract_context.name, asset_name, &asset) {
+            Err(Error::Runtime(RuntimeErrorType::NoSuchAsset, _)) => Ok(()),
+            Ok(_owner) => return Ok(Value::error(Value::Int(1))),
```

Comment:
The error code should be a named `const` instead of a magic value, IMHO.

Reply:
Addressed in 108f339e